### PR TITLE
[AV-86224] Generated docs for Terraform V1.3.0

### PR DESCRIPTION
**Pull request by a-atri**
## Jira

* AV-86224

## Description

Please include a summary of the fix/feature/change, including any relevant motivation and context.

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [x] Documentation fix/enhancement

### How was this change tested and do you have evidence? _**(REQUIRED: Select at least 1)**_

- [ ] Manually tested
- [ ] Unit tested
- [ ] Acceptance tested
- [x] Unable to test / will not test (Please provide comments in section below)

### Testing

<details open>
  <summary>Testing</summary>
  
</details>

## Required Checklist:

- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if required)
- [x] I have run make fmt and formatted my code
- [x] I have made sure that no schema field is marked with both requiresReplace and computed
## Further comments

### CODE REVIEW

File: docs/guides/1.3.0-upgrade-guide.md
Comment (rajagp) [inline docs/guides/1.3.0-upgrade-guide.md:13]: @a-atri  @aniket-Kumar-c  What about system events and activity logs?

File: docs/guides/1.3.0-upgrade-guide.md
Comment (rajagp) [inline docs/guides/1.3.0-upgrade-guide.md:98]: @PaulomeeCb  Curious: I wonder how one will use this functionality in an automated environment with Terraform. The command needs to be run out of band before the rest of scripts can be executed. Should we provide some guidance here?

**Review by rajagp — CHANGES_REQUESTED** [review]
Maybe I missed changes to index.md. Could not find in PR We should update the example usage 
https://registry.terraform.io/providers/couchbasecloud/couchbase-capella/latest/docs#example-usage-2

File: docs/guides/1.3.0-upgrade-guide.md
Comment (PaulomeeCb) [reply docs/guides/1.3.0-upgrade-guide.md:98]: @rajagp I didn't work on the private endpoints actually, @l0n3star would be able to address this better.

File: docs/guides/1.3.0-upgrade-guide.md
Comment (PaulomeeCb) [reply docs/guides/1.3.0-upgrade-guide.md:21]: @rajagp There are no other deprecations. Also, just FYI `configurationType` is deprecated in the api, and we removed the field from Terraform, it doesn't exist in the provider anymore.

File: docs/guides/1.3.0-upgrade-guide.md
Comment (l0n3star) [reply docs/guides/1.3.0-upgrade-guide.md:98]: @rajagp Good question.  Short answer, this command is not particularly useful in an automated environment.  

Long answer.  Assume customers will configure VPC endpoints with terraform as well.  They would have to use the CSP's terraform provider.  For example with AWS it might look like this:

```
resource "aws_vpc_endpoint" "example" {
  vpc_id            = var.vpc_id
  service_name      = "com.amazonaws.vpce.us-east-1.vpce-svc-1234"
  vpc_endpoint_type = "Interface"
  subnet_ids        = var.subnet_ids
}
```

Customers will have to supply `vpc_id` and `subnet_ids`.   `vpc_endpoint_type` must be `Interface`.  What customers need from us is `service_name`.  It would be better if we just returned the service name so it can be referenced in AWS's resource block.

File: docs/guides/1.3.0-upgrade-guide.md
Comment (rajagp) [reply docs/guides/1.3.0-upgrade-guide.md:21]: @PaulomeeCb  OK- Looks like we did't capture that in release notes anywhere :-) ...luckily sounds like no one was impacted. lets add example for single node (I think some user asked for that on how to do single node deployments)

File: docs/guides/1.3.0-upgrade-guide.md
Comment (rajagp) [reply docs/guides/1.3.0-upgrade-guide.md:98]: @l0n3star I think we should add an enhancement for 1.3.1 to return the `service_name`- we can do that in provider right?

File: docs/guides/1.3.0-upgrade-guide.md
Comment (PaulomeeCb) [reply docs/guides/1.3.0-upgrade-guide.md:21]: Yes that's true :) Let me add this example to this PR

Comment (PaulomeeCb) [reply docs/guides/1.3.0-upgrade-guide.md:21]: Added example, commit - https://github.com/couchbasecloud/terraform-provider-couchbase-capella/pull/220/commits/f2f5bd9052090316f8eb208209e5fb89f4607dda

File: docs/guides/1.3.0-upgrade-guide.md
Comment (l0n3star) [reply docs/guides/1.3.0-upgrade-guide.md:98]: We can do it in the provider.  Just need to extract `service_name` from what's returned by the API.  

And of course need to do something similar for Azure.

File: docs/guides/1.3.0-upgrade-guide.md
Comment (a-atri) [reply docs/guides/1.3.0-upgrade-guide.md:13]: @rajagp Added the system events and activity logs in the list

**Comment by a-atri** [comment]
@rajagp Updated the Index.md file

**Review by Rebecca-Martinez007 — COMMENTED** [review]
Just a few wording/punctuation changes. Waiting on a quick look over by @simon-dew. Should be good to go after that.

**Review by simon-dew — COMMENTED** [review]
The repeated text "see below for nested schema" looks generated. Is it configurable at all? I have some minor thoughts about the capitalization and punctuation of that phrase, but it's no blocker.

Other than that, I have nothing to add to @Rebecca-Martinez007 's comments

File: docs/guides/1.3.0-upgrade-guide.md
Comment (a-atri) [reply docs/guides/1.3.0-upgrade-guide.md]: @rajagp Added the depreciation message for the 'configurationType' field

File: docs/guides/1.3.0-upgrade-guide.md
Comment (Rebecca-Martinez007) [inline docs/guides/1.3.0-upgrade-guide.md:22]:
```suggestion
* The `configurationType` attribute in the cluster resource is now deprecated.
```

**Review by Rebecca-Martinez007 — APPROVED** [review]
Just a minor nit with the new sentence added for the deprecated field. Not a blocker, just a small suggestion. Thank you @a-atri !

Simon also left a small comment about the " (see [below for nested schema] " sentence.

**Review by Rebecca-Martinez007 — DISMISSED** [review]
@simon-dew we need to get this approval out, but @a-atri said we'd incorporate your feedback in the next Terraform release. I'm giving the approval here to move forward.

**Review by SaicharanCB — APPROVED** [review]
+1

**Review by Lagher0 — DISMISSED** [review]

**Review by nishant-tripathy — APPROVED** [review]
LGTM

**Review by Lagher0 — APPROVED** [review]

File: docs/guides/1.3.0-upgrade-guide.md
Comment (l0n3star) [reply docs/guides/1.3.0-upgrade-guide.md]: @rajagp 

On second thought I think regex is a bad idea.  What if the script changes and this breaks regex ?  A robust solution is to just return service name from the endpoint.  Luckily this is available to us from `network` component, so pretty easy to change V4 API.

**Pull request merged**
